### PR TITLE
Fixes for Tag&Probe in 7.4.X (and later)

### PR DESCRIPTION
**Pull request by gpetruc**
Fix a couple of issues needed to get tag and probe running in 7.4.X release.
Tested for muons using 74X https://github.com/cms-analysis/MuonAnalysis-TagAndProbe (after PR 2)
Not terribly clear whether it's ok also from electron side

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@cmsbuild 
the comparisons are marked as "queued" for more than 3 hours now.
Please check the system.
I suppose the clang issue hasn't blocked the comparisons

**Comment by nclopezo** [comment]
Hi Slava, 
The comparison had an issue when trying to clone the github repository. I restarted it.

File: PhysicsTools/TagAndProbe/plugins/AnythingToValueMap.h [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    private:
+      bool failSilently_;
+      edm::InputTag src_;
+      Adaptor adaptor_;
+  };
+
+template<class Adaptor, class Collection, typename value_type>
+void AnythingToValueMap<Adaptor,Collection,value_type>::produce(edm::Event & iEvent, const edm::EventSetup & iSetup) {
+    edm::Handle<Collection> handle;
+    iEvent.getByLabel(src_, handle);
```
Comment (slava77) [inline PhysicsTools/TagAndProbe/plugins/AnythingToValueMap.h]: no consumes?
or is it supposed to happen in the Adaptor somewhere?

Comment (gpetruc) [inline PhysicsTools/TagAndProbe/plugins/AnythingToValueMap.h]: indeed, I have not ported it to consumes.
I suppose I can do and update the PR.

File: PhysicsTools/TagAndProbe/plugins/ProbeMulteplicityProducer.cc [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+
+void 
+ProbeMulteplicityProducer::produce(edm::Event & iEvent, const edm::EventSetup & iSetup) {
+    using namespace edm;
+
+    // read input
+    Handle<View<reco::Candidate> > pairs;
+    iEvent.getByLabel(pairs_,  pairs);
```
Comment (slava77) [inline PhysicsTools/TagAndProbe/plugins/ProbeMulteplicityProducer.cc]: consumes should be declared

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/8801	a61a226
- the change in RECO parts is trivial
- jenkins tests run fine and comparisons with the baseline show no difference

I commented on some parts of the analysis code: as far as I understand that part of the code is not running in the standard workflows, so, the impact at this point is small. Still, that better be updated

**Comment by gpetruc** [comment]
Fixed the consumes migration. 

Now I'm trying `scram b vclean && scram build -k -j 8 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile' ` to see if I can reproduce the compilation errror cmsbuild complains about.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/8801	98cd7c286
the last changes have affected only PhysicsTools/TagAndProbe, not affecting the earlier reco signature for https://github.com/cms-sw/cmssw/pull/8801	a61a226

I double-checked that the jenkins tests comparisons show no differences as expected

**Comment by slava77** [comment]
@gpetruc please comment if we should expect a fix with the merge conflict here soon.
Thank you.
[we also need to figure out how to not lose PRs for this long]

**Comment by gpetruc** [comment]
Hi Slava,

I will do, but probably by the end of the week.

Giovanni

**Comment by gpetruc** [comment]
I would rather not push the successful merge to this same branch since this one currently merges fine in 74X while the one fixing the merge conflict in 75X will then by design merge only in 75X.
Unless you have strong objections, I'll then close this and open a new PR from the new branch.

**Comment by slava77** [comment]
@gpetruc Giovanni, pick what's the easiest for you.

**Comment by gpetruc** [comment]
Ok, it's now in #9512

**Pull request closed without merging**